SAT/ProofProducingSATSolver.hpp

```diff
@@ -73,61 +67,25 @@ class ProofProducingSATSolver final : public SATSolver {
     return _inner->failedAssumptions();
   }
 
-  SATClause* getRefutation()
-  {
-    // connect the added clauses ...
-    SATClauseList* prems = _addedClauses;
-
-    // ... with the current assumptions
-
-    // TODO: the assumption set will be empty after a call to solveUnderAssumptions()
-    // This does not matter much since refutations are only ever passed to collectFOPremises
-    // and there are no FO premises of assumption inferences
-
-    // So the below is commented out to prevent useless leaking
-
-    /*
-    for (size_t i=0; i < _assumptions.size(); i++) {
-      SATClause* unit = new(1) SATClause(1);
-      (*unit)[0] = _assumptions[i];
-      unit->setInference(new AssumptionInference());
-      SATClauseList::push(unit,prems);
-    }
-    */
-
-    _refutationInference->setPremises(prems);
-
-    return _refutation;
+  SATClauseList *minimizePremises(SATClauseList *premises) override {
+    return _inner->minimizePremises(premises);
   }
 
-  SATClauseList* getRefutationPremiseList() {
-    return _canMinimize ? _addedClauses : nullptr;
-  }
+  // all premises ever added
+  SATClauseList *premiseList() const { return _addedClauses; }
+
+  /*
+   * only those premises required to get unsat from the last solve() call
+   * may be a superset of those actually necessary depending on
+   * how well the minimisation process goes
+   */
+  SATClauseList *minimizedPremises() { return _inner->minimizePremises(_addedClauses); }
```

Comment:
It's not clear to me here, why:
1) we need the overload (not sure I understand the comment. Is the first line a separate sentence or should I really read: "only those premises ... may be a superset"? Overall, it feels the most obvious first line of the description of the function is missing)
2) why is it separated from its namesake by `premiseList`

Reply:
`ProofProducingSATSolver` is now just a wrapper around a SAT solver that remembers what was asserted. There is `minimizePremises` to conform with `SATSolver`, but I can't imagine when you would call that. `premiseList` gives you the list of all assertions. Then there is `minimize*d*`Premises`, which calls `minimizePremises` on the internally-kept list of premises. I'll fix the comment.